hls4ml/writer/vivado_accelerator_writer.py

```diff
@@ -87,6 +99,11 @@ def write_axi_wrapper(self, model):
                 newline = 'void {}_axi(\n'.format(model.config.get_project_name())
             elif '//hls-fpga-machine-learning insert include' in line:
                 newline = '#include "{}_axi.h"\n'.format(model.config.get_project_name())
+            elif config_weights and '//hls-fpga-machine-learning insert weights' in line:
+                newline = ''
+                for v in model.get_weight_variables():
+                    newline += indent + ', model_axi_t {name} [{shape}]\n'.format(name=v.name, shape=v.data_length)
+                newline += indent + ', char load_weights'
```

Comment:
Why not `bool` for `load_weights`?

Reply:
It came from a previous iteration. For a cleaner/safer C++ code, `bool` is the right choice. In hardware, it would become a single-bit register if that was just in the module, while on an AXI interface you can only use 8, 16, 32, etc sized words.